### DISCUSSION TITLE
Can this repo perform real-time voice changing?

### DISCUSSION THREAD

**Discussion by BenWeekes** [coqui-ai/TTS#1343] [General Q&A]
Is it possible to have a person who is speaking LIVE to have his voice changed (swapped) in real time such that the listener hears them as someone else?

**Comment by thorstenMueller** [comment]
If you mean by "live" and real time a (_nearly_) lipsync voice replacement then - no.
At minimum it will take the seconds to do STT stuff and then pipe that result to the TTS process. How long TTS will take depends on the model you choose (RTF).

**Reply by BenWeekes** [reply]
Thanks, that is what I thought. I had the idea of using image swapping ML (deep fake) on the mel spectogram for a real time conversion.

**Reply by DonFlymoor** [reply]
It could work with some seconds of latency
